glusterd2/transaction/context.go

```diff
@@ -84,24 +84,18 @@ func (c *Tctx) WithPrefix(prefix string) *Tctx {
 // Set attaches the given key-value pair to the context.
 // If the key exists, the value will be updated.
 func (c *Tctx) Set(key string, value interface{}) error {
-	json, e := json.Marshal(value)
-	if e != nil {
-		c.log.WithFields(log.Fields{
-			"error": e,
-			"key":   key,
-		}).Error("failed to marshal value")
-		return e
+	json, err := json.Marshal(value)
+	if err != nil {
+		c.log.WithError(err).WithField("key", key).Error("failed to marshal value")
+		return err
 	}
 
 	storeKey := c.prefix + "/" + key
-	_, e = store.Store.Put(context.TODO(), storeKey, string(json))
-	if e != nil {
-		c.log.WithFields(log.Fields{
-			"error": e,
-			"key":   storeKey,
-		}).Error("failed to set value")
+	_, err = store.Store.Put(context.TODO(), storeKey, string(json))
+	if err != nil {
+		c.log.WithError(err).WithField("key", key).Error("failed to set key in transaction context")
```

Comment:
I think here the key is being set in the store and not in transaction context

Reply:
That's right. I want to cleanup this repetitive pattern:

```go
if err := txn.Set(); err != nil {
    log.(....)
    return err
}
```

to 

```go
if err := txn.Set(); err != nil {
    return err
}
```

It reduces code clutter and greatly reduces lines of codes.

Reply:
so where are you gonna log the errors, in the calling function?

Reply:
Inside `txn.Set()`. Like done here in the PR